File: Medium/2026-02-06-3634-Minimum-Removals-to-Balance-Array/solution.js
var minimumRemovals = function(nums, k) {
    nums.sort((a, b) => a - b);
    let left = 0, result = nums.length;

    for (let right = 0; right < nums.length; right++) {
        while (nums[right] > nums[left] * k) {
            left++;
        }
        result = Math.min(result, right - left + 1);
    }

    return nums.length - result;
};

File: Medium/2026-02-06-3634-Minimum-Removals-to-Balance-Array/solution.py
class Solution:
    def minimumRemovals(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)
        left = 0
        result = n
        
        for right in range(n):
            while nums[right] > nums[left] * k:
                left += 1
            result = min(result, n - (right - left + 1))
        
        return result